pkg/scheduler/schedule_one_test.go

```diff
@@ -411,15 +411,16 @@ func TestSchedulerMultipleProfilesScheduling(t *testing.T) {
 		return true, binding, nil
 	})
 	controllers := make(map[string]string)
-	stopFn := broadcaster.StartEventWatcher(func(obj runtime.Object) {
+	if stopFn, err := broadcaster.StartEventWatcher(func(obj runtime.Object) {
 		e, ok := obj.(*eventsv1.Event)
 		if !ok || e.Reason != "Scheduled" {
 			return
 		}
 		controllers[e.Regarding.Name] = e.ReportingController
 		wg.Done()
-	})
-	defer stopFn()
+	}); err == nil {
```

Comment:
I don't think that's the right approach.
In case of error, the handler passed to StartEventWatcher will never be called, and thus wg.Done() will never be called. So the test will always timeout.

Instead, we should simply fail fast in case of `err != nil`

Reply:
Similar comment applies to other places.

Reply:
Thanks for the comments. I have fixed them. PTAL